forecasting/src/autogluon/forecasting/task/forecasting/dataset.py

```diff
@@ -0,0 +1,194 @@
+    0            A           1           2           3
+    1            C           1           2           3
+    2            B           1           2           3
+
+    """
+    if index_column is not None:
+        target = X.drop(index_column, axis=1)
+    else:
+        target = X.copy()
+    target_values = target.values
+    date_list = target.columns
+    processed_X = ListDataset([
+        {
+            FieldName.TARGET: target,
+            FieldName.START: pd.Timestamp(date_list[0], freq=freq),
+        }
+        for (target,) in zip(target_values, )
+    ], freq=freq)
+    return processed_X, freq
+
+
+class TimeSeriesDataset:
+
+    def __init__(self, dataset_name=None, train_path=None, test_path=None, is_train=True, prediction_length=None,
+                 index_column="index", time_column="date", target_column="target"):
+        """
+        dataset used for Forecasting Task
+
+        dataset_name: you can choose a built-in dataset in gluon-ts, default is the "electricity" dataset in gluon-ts
+
+        train_path: path to the train data
+
+        test_path: path to the test data, if not specified while train_path is specified, automatic split will be performed
+
+        is_train: whether the dataset is used for training, splitting will only happen when is_train == True
+
+        prediction_length: prediction_length for the forecasting task, need to be specified for automatic splitting
+
+        index_column: time series index, in the above example, there are three ts: A, B, C
+
+        time_column: date of a data
+
+        target_column: the predict target
+        """
+        self.is_train = is_train
+        if dataset_name is not None and dataset_name not in gluonts_builtin_datasets():
+            raise ValueError("Dataset {} is not available in gluonts.\n"
+                             " Current available ones:\n"
+                             "{}".format(dataset_name, gluonts_builtin_datasets()))
+
+        if dataset_name is not None:
+            if dataset_name not in gluonts_builtin_datasets():
+                raise ValueError("Dataset {} is not available in gluonts.\n"
+                                 " Current available ones:\n"
+                                 "{}".format(dataset_name, gluonts_builtin_datasets()))
+            try:
+                dataset = get_dataset(dataset_name)
+            except:
+                dataset = get_dataset(dataset_name, regenerate=True)
+            self.is_train = True
+            self.train_ds = dataset.train
+            self.test_ds = dataset.test
+            self.freq = dataset.metadata.freq
+            self.prediction_length = dataset.metadata.prediction_length
+        else:
+            if train_path is None:
+                raise ValueError("You must specify one of train_path and dataset_name to create TimeSeriesDataset.")
+            train_csv, freq = rebuild_tabular(X=pd.read_csv(train_path),
```

Comment:
Instead of `pd.read_csv`, consider using: `load_pd.load()`

https://github.com/awslabs/autogluon/blob/master/core/src/autogluon/core/utils/loaders/load_pd.py#L15

This method can be better in terms of memory usage and can read from s3 instead (as well as multi-part files)